sdk/runtime/runenv.go

```diff
@@ -102,14 +103,26 @@ type RunEnv struct {
 	RunParams
 	*logger
 
-	unstructured chan *os.File
-	structured   chan *zap.Logger
+	MetricsPusher *push.Pusher
+	unstructured  chan *os.File
+	structured    chan *zap.Logger
 }
 
 // NewRunEnv constructs a runtime environment from the given runtime parameters.
 func NewRunEnv(params RunParams) *RunEnv {
+	containerName, _ := os.Hostname()
 	re := &RunEnv{
-		RunParams:    params,
+		RunParams: params,
+		MetricsPusher: push.New("http://prometheus-pushgateway:9091", "testground/plan").
```

Comment:
@coryschwartz on a second thought - how is this actually working? I would expect for testplan instances to not have route to `prometheus-pushgateway` by default, only be able to resolve it on DNS. I don't see changes to the `sidecar`, so `prometheus-pushgateway` would be resolved correctly, but I am not sure why individual testplans would have a route to the IP of the `pushgateway`?

Then again I did run this with https://github.com/ipfs/testground/pull/616 and was getting metrics in Prometheus...

For the `influxdb` example, I had to add route to influxdb: see https://github.com/ipfs/testground/pull/605/files#diff-f6eeba9855b7c2b6fb2f75fcfcd26a48

Reply:
it's unroutable to the sidecar, but for the plan containers, the service resource created is within a routable range.